### PR TITLE
0.2.1 batch norm fixes

### PR DESCRIPTION
**Pull request by samdow**
Adds two commits for batch norm bug fixes. From discussion on #273 and #867

cc @zou3519 I ended up cherry-picking two commits since #958 also caught a typo and need for a contiguous call. However, lines-wise, this ends up being a lot of testing changes. I think I can refactor both changes to have fewer testing changes but will end up looking different than the version on main. Any thoughts as to if I should keep it like this or try to refactor for fewer test changes?

### CODE REVIEW

**Comment by samdow** [comment]
Change: since it's failing so many tests I'm going to go with option 1 and try to re-cherry-pick with very minimal testing changes

**Comment by zou3519** [comment]
> Change: since it's failing so many tests I'm going to go with option 1 and try to re-cherry-pick with very minimal testing changes

Sounds good

**Comment by zou3519** [comment]
I wouldn't worry about the mac signal -- the macos signal is failing because they deprecated the xcode image I think

**Review by zou3519 — APPROVED** [review]
code lgtm, let's wait for tests to pass

**Comment by samdow** [comment]
Circle CI shows passing for [all jobs but mac](https://app.circleci.com/pipelines/github/pytorch/functorch/3649/workflows/b26a7c68-58fc-4bde-8371-f89c1a4da48d). Looks like github checks aren't updating
